### PR TITLE
Parse.File getData

### PR DESCRIPTION
**Pull request by dplewis**
Closes: https://github.com/parse-community/Parse-SDK-JS/issues/678

Adds similar functionality for `.getData()` function in the [Parse SDK for Android](https://parseplatform.org/Parse-SDK-Android/api/com/parse/ParseFile.html#getData--) and [Parse SDK for PHP](http://parseplatform.org/parse-php-sdk/classes/Parse.ParseFile.html#method_getData)

Returns base64 (not full base64 with `data:...`)

### CODE REVIEW

File: src/ParseFile.js [hunk 1]
```diff
@@ -189,10 +212,19 @@ class ParseFile {
...
-        this._previousSave = controller.saveUri(this._name, this._source, options).then((res) => {
+        this._previousSave = controller.download(this._source.uri).then((result) => {
+          const newSource = {
+            format: 'base64',
+            base64: result.base64,
+            type: result.contentType,
+          };
+          this._data = result.base64;
+          return controller.saveBase64(this._name, newSource, options);
+        }).then((res) => {
```
Comment (acinader) [inline src/ParseFile.js]: I know it isn't your code, but I'm surprised that this function doesn't always return a promise!  

if !this._previousSave, then this wont return a promise and anything that then's this func will error.

maybe in practive there is always a previous state when this is called???

Comment (dplewis) [reply src/ParseFile.js]: I don't know why thats there. Been there since the beginning of the SDK. But this always returns a Promise.

Comment (dplewis) [reply src/ParseFile.js]: Want me to clean it up or leave it?

Comment (acinader) [reply src/ParseFile.js]: definitely your call

the class could definitely use some factoring out in any number of places (base64 comma location stuff for instance), but your changes look good to me and it works as is.

**Pull request merged**